vault-plugins/flant_iam/backend/path_role.go

```diff
@@ -75,8 +77,15 @@ func (b roleBackend) paths() []*framework.Path {
 		// List
 		{
 			Pattern: "role/?",
+			Fields: map[string]*framework.FieldSchema{
+				"show_archived": {
+					Type:        framework.TypeBool,
+					Description: "Option to list archived roles",
+					Required:    false,
+				},
+			},
 			Operations: map[logical.Operation]framework.OperationHandler{
-				logical.ListOperation: &framework.PathOperation{
+				logical.ReadOperation: &framework.PathOperation{
```

Comment:
Interesting, why list operation does not fit here. Maybe we should have a comment about it?

Reply:
ListOperation in vault cut every params you want to pass. In new LifeCycle we everywhere need pass param "show_archived", so we need replace ListOperation for ReadOperation.